### ISSUE TITLE
Scala: //Find the last but one element of a list.

### ISSUE THREAD

**Issue by Akhileshcloud7** [TheDigitalCatOnline/thedigitalcatonline.github.com#20]
I'm trying to print last but one element of the list, this is the code snippet here. But when I ran this code, code is not giving any output on console though the program executes successfully, and when I tried to put println() statement its throwing error with return type of function. Please find the below output and help to get the output of code
 
``` scala
object P02 extends App {
    //Find the last but one element of a list.
    val l=List(4,5,3,9,1)
    def penultimatEele[A](l:List[A]):A = {
        if (l.isEmpty) throw new NoSuchElementException
        l.init.last
    }
    def penult[A](n:Int, l:List[A]):A={
        if (n<=0) throw new IllegalArgumentException
        if (n>l.length) throw new NoSuchElementException
        l.takeRight(n).head
    }
}
```

**Comment by lgiordani** [comment]
The problem is that you are never calling the function that you defined. any `def` creates a function, it's like defining a process and giving it a name, but you have to _call_ the function, that is to execute it on some data, in this case what you want is to call it on the `val l` that you defined just above. Have you seen my answer for the previous PR that you closed #19 ? In the REPL log I called the function that you defined, and you have to do the same here. Let me know if this is clear enough

**Comment by Akhileshcloud7** [comment]
Thanks @lgiordani , i understand the topic. Thanks for your help
